src/v/rpc/transport.cc

```diff
@@ -88,9 +88,8 @@ ss::future<> base_transport::do_connect(clock_type::time_point timeout) {
           "cannot do_connect with a valid connection. remote:{}",
           server_address()));
     }
+    auto socket = ss::engine().net().socket();
```

Comment:
add code comment that `ss:;engine().socket()` doesn't throw & we need that to cleanup seastar resources.